test_vm/src/lib.rs

```diff
@@ -50,12 +50,9 @@ mod messaging;
 pub use messaging::*;
 
 /// An in-memory rust-execution VM for testing builtin-actors that yields sensible stack traces and debug info
-pub struct TestVM<'bs, BS>
-where
-    BS: Blockstore,
-{
+pub struct TestVM<'bs> {
     pub primitives: FakePrimitives,
-    pub store: &'bs BS,
+    pub store: &'bs TrackingMemBlockstore,
```

Comment:
The MockVM uses an `Rc<>` vs TestVM using an `& 'bs`. Probably one of these patterns is better than the other. The `Rc<>` is flexible and removes some lifetime specifier noise from the code. Maybe we should do that here too? @Stebalien wdyt?

Reply:
`Rc` is generally fine, especially because the `Blockstore` trait is implemented for `Rc<impl Blockstore>` (so there's no need to manually call `.borrow()` everywhere.

And dropping the lifetime would also be nice.

Reply:
Ah, wait. I'm confusing `RefCell` and `Rc`. Yeah, there's no reason not to use `Rc<TestBlockstore>`.

Reply:
I went down this path but ended up running into some really weird Rust compiler errors. Are you okay doing this in a follow up PR ?

For now, I'd like to focus on and resolve https://github.com/filecoin-project/builtin-actors/pull/1445#discussion_r1362020250. Please can you leave your thoughts there ?